index.js

```diff
@@ -1 +1,48 @@
-console.log('hello world');
+import express from 'express';
+import { read } from './jsonFileStorage.js';
+
+const app = express();
+
+const handleRecipeByIndex = (req, res) => {
+  if (isNaN(req.params.index)) {
+    res.status(404).send('Sorry, we cannot find that!');
+    return;
+  }
+
+  read('data.json', (err, data) => {
+    const recipe = data.recipes[req.params.index];
+
+    if (recipe) res.send(recipe);
+    else res.status(404).send('Sorry, we cannot find that!');
+  });
+};
+app.get('/recipe/:index', handleRecipeByIndex);
+
+const handleRecipeByYield = (req, res) => {
+  if (isNaN(req.params.portion)) {
+    res.status(404).send('Sorry, we cannot find that!');
+    return;
+  }
+
+  const portion = parseInt(req.params.portion, 10);
+  read('data.json', (err, data) => {
+    const recipes = data.recipes.filter((recipe) => recipe.yield === portion);
+
+    if (recipes.length > 0) res.send(recipes);
+    else res.status(404).send('Sorry, we cannot find that!');
+  });
+};
+app.get('/yield/:portion', handleRecipeByYield);
+
+const handleRecipeByLabel = (req, res) => {
+  const label = req.params.label.replaceAll('-', ' ');
+  read('data.json', (err, data) => {
+    const recipes = data.recipes.filter((recipe) => recipe.label.toLowerCase() === label);
+
+    if (recipes.length > 0) res.send(recipes);
+    else res.status(404).send('Sorry, we cannot find that!');
```

Comment:
this is rather subjective but if you're using an `if else` statement, in terms of readability, it helps to separate them with the curly brackets instead. You could also consider a ternary operator too since there are only two condition. read more about it here

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

```suggestion
    if (recipes.length > 0) {  
         res.send(recipes); 
     } else {
         res.status(404).send('Sorry, we cannot find that!');
     }
```

Reply:
How do you use a ternary operator on this if-else? 
The "if" uses only res.send() but the "else" uses res.status().send().
 I could go 
` res.status((recipes.length>0) ? 200 : 404).send((recipes.length>0) ? recipes : 'Sorry, we cannot find that')
` 
but the double conditional doesn't feel right.
 
I sometimes use ternary operator if the line includes assignment to the same variables, and the line doesn't go over eslint's 100 length limit.

Reply:
For the curly bracket suggestions, sometimes I like to go without them when it's only a one-liner. But I get what you said about readability concern.

Reply:
you could consider something like this

```suggestion
    recipes.length > 0 ? response.send('has recipes!') : response.status(404).send('no recipes!');
```

Reply:
Did not realize you can use tertiary operator like that! 
I thought it needs to be used inside a parentheses like for function parameters, or on the right side of an assignment.
Thanks!